File: cell.py
class cell:
  def __init__(self, row, col):
    self.row = row
    self.col = col
  
  def isEqual(self, otherCell):
    return self.row==otherCell.row and self.col==otherCell.col
  
  def __str__(self):
    return "CELL: " + str(self.row) + ", " + str(self.col)

File: cell/cell.cpp
#include <cell.hpp>
#include <cmath>

cell::cell() {
  row = -1;
  col = -1;
}

cell::cell(int r, int c) {
  row = r;
  col = c;
}

double cell::euclidean_distance(cell& other) {
  if(other.row<0 || other.col<0)
    return 100000000;
  return sqrt(pow(other.row-row, 2) + pow(other.col-col, 2));
}

bool cell::is_equal(cell& other) {
  return (row==other.row && col==other.col);
}